### ISSUE TITLE
"Overrides value from a preceding case" false positive

### ISSUE THREAD

**Issue by reedy** [kalessil/phpinspectionsea#1298]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) v3.0.13 |
| Language level | 7.0                       |

Current behaviour
--
Given code

```lang=php
		switch ( $case ) {
			case 'instrumental': # instrumental
				$word = 'z ' . $word;
			case 'lokatiw': # lokatiw
				$word = 'wo ' . $word;
				break;
		}
```

The assignment in second case is marked as overriding, but it's not a complete override, it's a prepend

Expected behaviour
--
If the code was
```lang=php
		switch ( $case ) {
			case 'instrumental':
				$word = 'z ' . $word;
			case 'lokatiw':
				$word = 'wo ';
				break;
		}
```

I could see the error being valid there; there's clearly a mistake. The code is unconditionally overwritten, so the first case and first assignment are useless

But in the case of the code in the current behaviour, we're just prepending (and appending should have the same behaviour), so we're not overriding

This code gives the same issue

```lang=php
		switch ( $case ) {
			case 'instrumental':
				$word = 'z ' . $word;
			case 'lokatiw':
				$word = $word . 'wo';
				break;
		}
```

But this code works as expected, with no issue flagged:

```lang=php
		switch ( $case ) {
			case 'instrumental':
				$word = 'z ' . $word;
			case 'lokatiw':
				$word .= 'wo';
				break;
		}
```

I guess this is a slightly different issue as the code does the same thing, although the syntax is slightly different (for the simpler append action), but gives a different (or in this case, no) error. PHPStorm still highlights no break on the first statement, which is fine

Environment details
---
PhpStorm 2019.1.1 Preview
Build #PS-191.6707.42, built on April 10, 2019
Licensed to MediaWiki / Sam Reed
Subscription is active until May 1, 2019
For non-commercial open source development only.
JRE: 1.8.0_202-release-1483-b44 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.4

**Comment by reedy** [comment]
<img width="738" alt="Screenshot 2019-04-17 at 23 28 53" src="https://user-images.githubusercontent.com/67615/56325256-755cff00-6169-11e9-8f5e-1a9e1e94aa31.png">

**Comment by reedy** [comment]
File is https://github.com/wikimedia/mediawiki/blob/master/languages/classes/LanguageDsb.php for reference

**Comment by orklah** [comment]
Hi!

I think it's the same issue than here: #1226

**Comment by reedy** [comment]
> Hi!
> 
> I think it's the same issue than here: #1226

Ah, looks to be basically the same. Slightly different usages, but the same result... And I've posted a few more use cases.examples :)

**Comment by kalessil** [comment]
Fixed!
